### PR TITLE
Fix errors and changes after importing add-ons in Eclipse

### PR DESCRIPTION
**Pull request by wborn**
* Add missing Eclipse project/classpath files
* Fix null analysis errors showing in Eclipse
* Remove ignored Eclipse .settings files

With these changes there are no unstaged changes and only one error showing in FileTailer (#5442) after importing the add-ons in Eclipse.

### CODE REVIEW

File: bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java [hunk 1]
```diff
@@ -116,24 +117,27 @@ public void stop() {
      *
      * @return a list of device ids
      */
-    public List<SensorId> getDirectory(String basePath) throws OwException {
+    public @NonNullByDefault({}) List<SensorId> getDirectory(String basePath) throws OwException {
```
Comment (cpmeister) [inline bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java:120]: Why was `@NonNullByDefault({})` added here?

File: bundles/org.openhab.binding.smartmeter/.classpath [hunk 2]
```diff
@@ -6,12 +6,6 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="src" output="target/classes" path="src/3rdparty/java">
-		<attributes>
-			<attribute name="optional" value="true"/>
-			<attribute name="maven.pomderived" value="true"/>
-		</attributes>
-	</classpathentry>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.smartmeter/.classpath:14]: This isn't needed?

File: bundles/org.openhab.binding.logreader/.classpath [hunk 3]
```diff
@@ -6,12 +6,6 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="src" output="target/classes" path="src/3rdparty/java">
-		<attributes>
-			<attribute name="optional" value="true"/>
-			<attribute name="maven.pomderived" value="true"/>
-		</attributes>
-	</classpathentry>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.logreader/.classpath:14]: not needed?

File: bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java:120]: I'd rather not use it but It has issues with captures:

`Null type mismatch (type annotations): required 'Collector<? super @Nullable SensorId,Object,@NonNull List<@NonNull SensorId>>' but this expression has type 'Collector<@NonNull SensorId,capture#of ?,List<@NonNull SensorId>>'`

File: bundles/org.openhab.binding.logreader/.classpath (hunk 3 above)
Comment (wborn) [reply bundles/org.openhab.binding.logreader/.classpath:14]: I fixed the logreader in the new commit https://github.com/openhab/openhab-addons/pull/7270/commits/b529604307c1a3016a8132cade3ba6f987842299

File: bundles/org.openhab.binding.smartmeter/.classpath (hunk 2 above)
Comment (wborn) [reply bundles/org.openhab.binding.smartmeter/.classpath:14]: I added it back again.

File: bundles/org.openhab.binding.logreader/.classpath [hunk 4]
```diff
@@ -6,12 +6,7 @@
...
+	<classpathentry kind="src" path="src/3rdparty/java"/>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.logreader/.classpath:9]: no output path?

File: bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java (hunk 1 above)
Comment (cpmeister) [reply bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java:120]: I guess it is better than casting the return value to `List<SensorId>`

File: bundles/org.openhab.binding.smartmeter/.classpath [hunk 5]
```diff
@@ -6,26 +6,22 @@
...
+	<classpathentry kind="src" path="src/3rdparty/java"/>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.smartmeter/.classpath:9]: no output path?

File: bundles/org.openhab.binding.logreader/.classpath (hunk 4 above)
Comment (wborn) [reply bundles/org.openhab.binding.logreader/.classpath:9]: It'll use the default which is `target/classes`

File: bundles/org.openhab.binding.smartmeter/.classpath (hunk 5 above)
Comment (wborn) [reply bundles/org.openhab.binding.smartmeter/.classpath:9]: It'll use the default which is `target/classes`

File: bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java:120]: Casting also seems to work. They are both workarounds and I don't really prefer one over the other. ;-)

**Comment by cweitkamp** [comment]
One more thing: There is a duplicate project name. The LinuxInput Binding calls itself ` org.openhab.binding.astro` instead of `org.openhab.binding.linuxinput`.

https://github.com/openhab/openhab-addons/blob/5cef8d3e110d317500e76c508d76e82f625d3514/bundles/org.openhab.binding.linuxinput/.project#L2-L4

Otherwise: Great job!

**Pull request merged**